File: ABC/254/D.py
n = int(input())
p = [i for i in range(n+1)]
for i in range(2, n+1):
    x = i*i
    if x > n:
        break
    for j in range(x, n+1, x):
        while p[j] % x == 0:
            p[j] //= x

cnt = [0] * (n+1)
for i in range(1, n+1):
    cnt[p[i]] += 1

res = 0
for i in range(n+1):
    res += cnt[i]*cnt[i]

print(res)

File: ABC/254/D.cpp
#include <iostream>
#include <vector>

using namespace std;
using ll = long long;

#define rep(i, n) for (int i = 0; i < (n); i++)

int main() {
  int n;
  cin >> n;
  vector<int> f(n+1);
  rep(i, n+1) f[i] = i;

  for (int i = 2; i <= n; i++) {
    int x = i * i;
    if (x > n) break;
    for (int j = x; j <= n; j += x) {
      while (f[j]%x == 0) f[j] /= x;
    }
  }
  vector<int> cnt(n+1);
  for (int i = 1; i <= n; i++) {
    cnt[f[i]]++;
  }

  ll res = 0;
  rep(i, n+1) res += (ll)cnt[i] * cnt[i];
  cout << res << endl;
}
